Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -93,15 +583,78 @@ theorem isEllDivSequence_id : IsEllDivSequence id :=
 variable {W}
 
 lemma IsEllSequence.smul (h : IsEllSequence W) (x : R) : IsEllSequence (x • W) :=
-  fun m n r => by
-    linear_combination (norm := (simp only [Pi.smul_apply, smul_eq_mul]; ring1)) x ^ 4 * h m n r
+  fun m n r ↦ show _ = _ by linear_combination
+    (norm := (simp only [Pi.smul_apply, smul_eq_mul]; ring1)) x ^ 4 * (show _ = _ from h m n r)
 
 lemma IsDivSequence.smul (h : IsDivSequence W) (x : R) : IsDivSequence (x • W) :=
-  fun m n r => mul_dvd_mul_left x <| h m n r
+  (mul_dvd_mul_left x <| h · · ·)
 
 lemma IsEllDivSequence.smul (h : IsEllDivSequence W) (x : R) : IsEllDivSequence (x • W) :=
   ⟨h.left.smul x, h.right.smul x⟩
 
+lemma IsEllSequence.map (h : IsEllSequence W) : IsEllSequence (f ∘ W) := by
+  simpa using (congr_arg f <| h · · ·)
+
+lemma IsDivSequence.map (h : IsDivSequence W) : IsDivSequence (f ∘ W) :=
+  (map_dvd f <| h · · ·)
+
+lemma IsEllDivSequence.map (h : IsEllDivSequence W) : IsEllDivSequence (f ∘ W) :=
+  ⟨h.1.map f, h.2.map f⟩
```

Comment:
Seems like these and everything below are unrelated to `_root_.IsEllSequence.of_oddRec_evenRec`, but instead a consequence of `rel₄_of_oddRec_evenRec` and the lemmas for `invarNum/Denom`, and only added here because it's part of a downstream PR. I hope this is not too much work, but given the size of the PR (and the novelty of the argument), I suggest that you split off everything that's not directly related to the main argument.